### ISSUE TITLE
Python 3.8 and Python/C++ mixed-mode debugging

### ISSUE THREAD

**Issue by naughtong** [microsoft/PTVS#6265]
I can use python 3.8 with visual studio 2019 (Version 16.7.4) but I run into problems when trying to use Python/C++ mixed-mode debugging.
It is surprising that Microsoft is taking so long to include Python 3.8, there were plenty of issues reported over a year ago for Python 3.8.

**Comment by huguesv** [comment]
We have updated the mixed-mode debugger to support Python 3.8.

What issue are you having?

These are the known issues related to mixed-mode debugging: https://github.com/microsoft/PTVS/issues?q=is%3Aopen+is%3Aissue+label%3A%22area%3ANative+Modules%22

**Comment by naughtong** [comment]
Thanks for letting me know, but I still get this message.

I took a couple of updates but I still get this message when I open a python solution:

"You're using Python 3.8. Some new features might not work as expected until Visual Studio officially supports this version. More info Don't show this message again"

I will try out the mixed mode debugging again in the morning. The errors were related to new Python 3.8 features.

**Comment by naughtong** [comment]
The 3.8 errors and warnings such as "Unexpected token ‘/’	operator.py" are gone. Although the banner at the top of Visual Studio "You're using Python 3.8. Some new features m ..." is still there.

The main imports in the test py files are a .pyd file and the associated .py file. The .pyd file was built in debug mode. I ran some tests on the command line using python_d and they work.

But if I run the same tests with debugging in Visual Studio 2019 it gets past loading the .pyd file but when loading the .py file it just exists. The output just says:

The program '[4064] pythonw_d.exe' has exited with code 1 (0x1).

**Comment by naughtong** [comment]
Unfortunately I just got an error for something that is new in python 3.8:

The error was "invalid token '/' "  in threading.py  def excepthook(args, /)

This is the version of Visual Studio that I am using:

Microsoft Visual Studio Community 2019
Version 16.7.5Microsoft Visual Studio Community 2019
Version 16.7.5

**Comment by huguesv** [comment]
The banner warning is displayed unconditionally when using Python 3.8 or later, regardless of the specific feature you are using, so that behavior is expected.

Since the editor does not support any of the Python 3.8 features yet, I also expect editor errors/warnings/squiggles when the code includes new 3.8 syntax.

But I'm surprised that this would cause the debugger to stop. We could have an issue in mixed-mode debugger with some incorrect Python object internal structure with regards to positional only arguments changes in 3.8. Our existing tests may have never hit a code path that used them.

**Comment by int19h** [comment]
If this is debugging *unit tests*, wouldn't it be using the regular debugger, not mixed-mode?

**Comment by huguesv** [comment]
Not necessarily, the test explorer debugger attaching code respects the "Enable native code debugging" option.

**Comment by naughtong** [comment]
This is a little background:

Quantlib is an open source quite large C++ financial analytical library. 
For the Python version, SWIG is used to created a C++ wrapper which creates a .pyd file and a Python py file (your import library).
There are Python examples and tests. 
From the command line they all work, whether built in debug mode or not.

In visual studio 2019 the release version runs without a problem and you can step through the python code.
The problems start when you try to use python_d and the debug .pyd file.
The Enable native code debugging is set.
